internal/workflow/processing.go

```diff
@@ -911,18 +916,41 @@ func (w *ProcessingWorkflow) preprocessing(ctx temporalsdk_workflow.Context, tin
 		WorkflowID:        fmt.Sprintf("%s-%s", w.cfg.Preprocessing.Temporal.WorkflowName, uuid.New().String()),
 		ParentClosePolicy: temporalapi_enums.PARENT_CLOSE_POLICY_TERMINATE,
 	})
-	var result preprocessing.WorkflowResult
+	var ppResult preprocessing.WorkflowResult
 	err = temporalsdk_workflow.ExecuteChildWorkflow(
 		preCtx,
 		w.cfg.Preprocessing.Temporal.WorkflowName,
 		preprocessing.WorkflowParams{RelativePath: relPath},
-	).Get(preCtx, &result)
+	).Get(preCtx, &ppResult)
 	if err != nil {
 		return err
 	}
 
-	tinfo.TempPath = filepath.Join(w.cfg.Preprocessing.SharedPath, filepath.Clean(result.RelativePath))
+	tinfo.TempPath = filepath.Join(w.cfg.Preprocessing.SharedPath, filepath.Clean(ppResult.RelativePath))
 	tinfo.IsDir = true
 
-	return nil
+	opts := withLocalActivityOpts(ctx)
+	var savePPTasksResult localact.SaveIngestTasksActivityResult
+	err = temporalsdk_workflow.ExecuteLocalActivity(
+		opts,
+		localact.SaveIngestTasksActivity,
+		localact.SaveIngestTasksActivityParams{
+			PkgSvc:               w.pkgsvc,
+			RNG:                  w.rng,
+			PreservationActionID: tinfo.PreservationActionID,
+			Tasks:                ppResult.IngestTasks,
+		},
+	).Get(opts, &savePPTasksResult)
+	if err != nil {
+		return err
+	}
+
+	switch ppResult.Outcome {
+	case preprocessing.OutcomeSystemError:
+		return errors.New("preprocessing workflow: system error")
+	case preprocessing.OutcomeContentError:
+		return errors.New("preprocessing workflow: validation failed")
+	default:
```

Comment:
Should we check for unexpected values or is the enum enough to verify that?

Reply:
Yes, good idea.